scm/scm_test.go

```diff
@@ -0,0 +1,232 @@
+package scm
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type fakeExecCommand struct {
+	id      string
+	execCmd func(command string, args ...string) *exec.Cmd
+	command string
+}
+
+func newFakeExecCommand(id string) *fakeExecCommand {
+	c := &fakeExecCommand{}
+	c.id = id
+	c.execCmd = func(name string, args ...string) *exec.Cmd {
+		c.command = fmt.Sprintf("%s %s", name, strings.Join(args, " "))
+		cs := []string{"-test.run=TestHelperProcess", "--", name}
+		cs = append(cs, args...)
+		cmd := exec.Command(os.Args[0], cs...)
+		cmd.Env = []string{"GO_WANT_HELPER_PROCESS=1", fmt.Sprintf("GO_TEST_MODE=%s", id)}
+		return cmd
+	}
+	return c
+}
+
+func TestNew(t *testing.T) {
+	t.Run("success with https url", func(t *testing.T) {
+		baseDir := os.TempDir()
+		srcURL := "https://github.com/screwdriver-cd/sd-local.git#test"
+
+		expected := &scm{
+			baseDir:   baseDir,
+			remoteURL: "https://github.com/screwdriver-cd/sd-local.git",
+			branch:    "test",
+		}
+
+		s, err := New(baseDir, srcURL)
+		defer os.RemoveAll(s.LocalPath())
+
+		expected.localPath = s.LocalPath()
+		assert.Equal(t, expected, s)
```

Comment:
I think it's better to assert only `s.baseDir`, `s.remoteURL` and `s.branch` than to do this

Reply:
I'll fix it.